Fix bug with initial facets in post filter

lbrant1 opened the pull request:
add method to allow overriding of is_initial

Commit:
Fix bug with initial facets in post filter

add method to allow overriding of is_initial

ruttenb commented on seeker/views.py:
```diff
@@ -748,6 +748,14 @@ def sort_descriptor(self, sort):
+        Can be overridden in case of non initial get requests that aren't ajax (submitted form)
+        This will depend on the seeker form html and the info coming in on the get request
+        """
+        return False if self.request.is_ajax() else True
```
return not self.request.is_ajax()

Commit:
simplify is_initial call

Commit:
fix for bug associated with overridng get_facet_data

post_filter_facet should just use the data that was selected by
get_facet_data and not the initial or get dictionary, in case the method
has been overridden
